test/test_nestedtensor.py

```diff
@@ -482,6 +482,76 @@ def test_clone(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, msg):
             nt1.clone(memory_format=torch.channels_last)
 
+    # cannot test torch.float16 because: RuntimeError: "bernoulli_scalar_cpu_" not implemented for 'Half'
+    @dtypes(torch.float, torch.double)
+    def test_dropout(self, device, dtype):
+        # edge case: empty nested tensor
+        nt0 = torch.nested_tensor([])
+        y = torch.nn.functional.dropout(nt0, 0.5)
+        self.nt_equal(nt0, y)
+        # normal nested tensor
+        ntensors = 4
+        nt = self.random_nt(device, dtype, ntensors, (4, 4))
+        # edge case: invalid dropout
+        self.assertRaises(ValueError, lambda: torch.nn.Dropout(-0.1))
+        self.assertRaises(ValueError, lambda: torch.nn.Dropout(1.1))
+        self.assertRaises(ValueError, lambda: torch.nn.functional.dropout(nt, -0.1))
+        self.assertRaises(ValueError, lambda: torch.nn.functional.dropout(nt, 1.1))
+        # edge case: no dropout
+        dropouter = torch.nn.Dropout(0.0)
+        y0 = dropouter(nt)
+        y1 = torch.nn.functional.dropout(nt, 0.0)
+        self.nt_equal(nt, y0)
+        self.nt_equal(nt, y1)
+        # edge case: all dropout
+        dropouter = torch.nn.Dropout(1.0)
+        y0 = dropouter(nt)
+        y1 = torch.nn.functional.dropout(nt, 1.0)
+        nt0 = nt.clone()
+        for i in range(ntensors):
+            nt0[i].fill_(0.0)
+        self.nt_equal(nt0, y0)
+        self.nt_equal(nt0, y1)
+        # normal case: normal dropout
+        p = 0.2
+        y = torch.nn.functional.dropout(nt, p)
+        expect = nt.clone()
+        for i in range(ntensors):
+            actual_tensor = y[i].view(-1)
+            expect_tensor = expect[i].view(-1)
+            for j in range(actual_tensor.shape[0]):
+                if actual_tensor[j].item() == 0.0:
+                    expect_tensor[j] = 0.0
+                else:
+                    expect_tensor[j] /= 1.0 - p
+        self.nt_equal(y, expect)
+        seed = torch.randint(0, 32767, (1,)).item()
+        torch.manual_seed(seed)
+        torch.cuda.manual_seed_all(seed)
```

Comment:
Does this CUDA-specific seed setting need to be here? I'd expect this to be handled internally by the test framework somehow. Does this no-op for `device='cpu'`?

Reply:
Maybe you can use `torch.testing._internal.common_utils.freeze_rng_state()` instead to accomplish what you need? I believe it works across devices